cdap-metadata-spi/src/main/java/io/cdap/cdap/spi/metadata/MetadataConsumerContext.java

```diff
@@ -34,4 +34,9 @@ public interface MetadataConsumerContext {
    * @return the MetadataConsumer properties for the program run
    */
   Map<String, String> getProperties();
+
+  /**
+   * Returns a {@link MetadataConsumerMetrics} object configured based on {@code context}.
+   */
+  MetadataConsumerMetrics getMetrics(Map<String, String> context);
```

Comment:
Can MetricsContext be used directly here? Is there a problem with introducing `cdap-watchdog-api` dependency to the SPI?

Reply:
I used this commit for reference - https://github.com/cdapio/cdap/commit/c90bcfae320b009b61eac368b607afa20b5ab419.
This is for a similar case, for dataproc runtime-spi. They have their own implementation of metrics here